server/src/main/java/io/crate/types/CharacterType.java

```diff
@@ -45,6 +60,93 @@ public class CharacterType extends StringType {
     public static final int ID = 27;
     public static final CharacterType INSTANCE = new CharacterType();
 
+    private static StorageSupport<Object> storageSupport(int lengthLimit) {
+        return new StorageSupport<>(
+            true,
+            true,
+            new EqQuery<Object>() {
+
+                @Override
+                public Query termQuery(String field, Object value, boolean hasDocValues, boolean isIndexed) {
+                    if (isIndexed) {
+                        return new TermQuery(new Term(field, BytesRefs.toBytesRef(stripAndPad(value, lengthLimit))));
+                    }
+                    if (hasDocValues) {
+                        return SortedSetDocValuesField.newSlowExactQuery(
+                            field, BytesRefs.toBytesRef(stripAndPad(value, lengthLimit)));
+                    }
+                    return null;
+                }
+
```

Comment:
Do I get it right, that it's same as StringType but with `stripAndPad` scattered around?

Can we instead add `stripAndPad` to the StringType, rename it to `maybeStripAndPad`, make it no-op for StringType and call everywhere like it's done here? 

Then CharacterType would have to only override maybeStripAndPad to make it no-op but would share same logic with StringType - probably would be helpful to propagate future changes/fixes to both types.

Reply:
No we cannot, because in String it's static thing, whereas in the character type it depends on the length limit. Additionally all this is called in an constructor, thus, you cannot create a member method (only static methods can be called) that can be overridden in the child class (CharacterType).

Reply:
You could extract a `StringEqQuery` with a `preprocess` function:

```java
final class StringEqQuery implements EqQuery<String> {

    private final UnaryOperator<String> preprocess;

    StringEqQuery(UnaryOperator<String> preprocess) {
        this.preprocess = preprocess;
    }

    ...
```

And then change the two StorageSupport constructions to use `new StringEqQuery(this::pad..)` and one with a identity function

Reply:
Thx @mfussenegger, could you please check? (maybe the whole PR)?